src/Handler/CurlFactory.php

```diff
@@ -125,6 +127,7 @@ public static function finish(callable $handler, EasyHandle $easy, CurlFactoryIn
     private static function invokeStats(EasyHandle $easy): void
     {
         $curlStats = \curl_getinfo($easy->handle);
+        assert(\is_array($curlStats));
```

Comment:
This can be fixed similarly to my other comment below, however it means that the potential type information about `total_time` is lost from being a `float` to `mixed`, while the assertion is not too nice, a better fix would be to have an array-shape for this to preserve that type information